### PR TITLE
Use javascript to set lastname, firstname when proxy depositing.

### PR DESCRIPTION
**Pull request by scherztc**

### CODE REVIEW

**Comment by hortongn** [comment]
I do think a feature test that adds a work as a delegate and checks that the resulting work's author is in lastname, firstname format would be good here.  See the test James did in spec/features/generic_works_spec.rb when he split names into first and last. https://github.com/uclibs/curate/commit/564267613c4d2e84bc4d262d56b42ba1c19c7551#diff-b400db19426bbdc118434e52388abe4dL139

**Comment by scherztc** [comment]
Sounds good.  We looked at putting it here:  https://github.com/uclibs/curate/blob/develop/spec/features/proxy_deposit_spec.rb

I will take care of that tomorrow.

**Comment by scherztc** [comment]
https://github.com/uclibs/scholar_uc/issues/672

File: spec/features/proxy_deposit_spec.rb [hunk 1]
```diff
@@ -36,7 +38,8 @@
       choose('Visible to the world.')
       check("I have read and accept the distribution license agreement")
       click_button("Create Article")
-      expect(page).to have_selector('#article_owner', text: user.name)
+      visit new_curation_concern_generic_work_path
```
Comment (jamesvanmil) [inline spec/features/proxy_deposit_spec.rb]: I'm not seeing what this line and the next are intending to do - it looks like we've just created a new Article, but we're going to create a new generic work before setting any spec expectations.

Same goes for for line 60 and 61 below.

**Comment by scherztc** [comment]
I was able to stream-line the test, per our conversation.  It checks for lastname, firstname immediately after selecting on behalf without form submission or extra route traversals.

**Pull request merged**